api/bucket.go

```diff
@@ -44,3 +46,17 @@ type (
 		Policy BucketPolicy `json:"policy"`
 	}
 )
+
+var validBucketExp = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$`)
+
+func (req BucketCreateRequest) Validate() error {
+	// make sure the bucket name complies with the restrictions for S3 transfer
+	// acceleration which are the regular S3 conventions with the additional
+	// restriction that the bucket name can't contain a period.
+	if strings.HasPrefix(req.Name, "xn--") ||
```

Comment:
Most examples online include this checks in the regex, like so `(?!(^xn--|.+-s3alias$))`. Is there a reason we check for prefix manually?